internal/dockercompose/event.go

```diff
@@ -30,79 +30,122 @@ func EventFromJsonStr(j string) (Event, error) {
 }
 
 // https://docs.docker.com/engine/reference/commandline/events/
-type Type string
+type Type int
 
 const (
-	// Add 'types' here (and to `UnmarshalJSON` below) as we support them
-	TypeContainer Type = "container"
+	// Add 'types' here (and to `stringToType` below) as we support them
+	TypeUnknown Type = iota
+	TypeContainer
 )
 
+var stringToType = map[string]Type{
+	"container": TypeContainer,
+}
+
+func (t Type) String() string {
+	for str, typ := range stringToType {
+		if typ == t {
+			return str
+		}
+	}
+	return "unknown"
+}
+
+func (t Type) MarshalJSON() ([]byte, error) {
+	s := t.String()
+	return []byte(fmt.Sprintf("%q", s)), nil
+}
+
 func (t *Type) UnmarshalJSON(b []byte) error {
 	s := string(b)
 	if unquoted, err := strconv.Unquote(s); err == nil {
 		s = unquoted
 	}
 
-	if s == "container" {
-		*t = TypeContainer
-	} else {
-		return fmt.Errorf("unknown `Type` in docker-compose event: %s", s)
-	}
+	typ := stringToType[s] // if type not in map, this returns 0 (i.e. TypeUnknown)
+	*t = typ
 	return nil
 }
 
-type Action string
+type Action int
 
 const (
-	// Add 'actions' here (and to `UnmarshalJSON` below`) as we support them
+	// Add 'actions' here (and to `stringToAction` below`) as we support them
 
 	// CONTAINER actions
-	ActionAttach     Action = "attach"
-	ActionCreate     Action = "create"
-	ActionDie        Action = "die"
-	ActionExecDie    Action = "exec_die"
-	ActionExecAttach Action = "exec_attach"
-	ActionExecCreate Action = "exec_create"
-	ActionKill       Action = "kill"
-	ActionRename     Action = "rename"
-	ActionRestart    Action = "restart"
-	ActionStart      Action = "start"
-	ActionStop       Action = "stop"
-	ActionUpdate     Action = "update"
+	ActionUnknown Action = iota
+	ActionAttach
+	ActionCommit
+	ActionCopy
+	ActionCreate
+	ActionDestroy
+	ActionDie
+	ActionExecCreate
+	ActionExecDetach
+	ActionExecDie
+	ActionExecStart
+	ActionExport
+	ActionHealthStatus
+	ActionKill
+	ActionOom
+	ActionPause
+	ActionRename
+	ActionResize
+	ActionRestart
+	ActionStart
+	ActionStop
+	ActionTop
+	ActionUnpause
+	ActionUpdate
 )
 
+var stringToAction = map[string]Action{
+	"attach":        ActionAttach,
+	"commit":        ActionCommit,
+	"copy":          ActionCopy,
+	"create":        ActionCreate,
+	"destroy":       ActionDestroy,
+	"die":           ActionDie,
+	"exec_create":   ActionExecCreate,
+	"exec_detach":   ActionExecDetach,
+	"exec_die":      ActionExecDie,
+	"exec_start":    ActionExecStart,
+	"export":        ActionExport,
+	"health_status": ActionHealthStatus,
+	"kill":          ActionKill,
+	"oom":           ActionOom,
+	"pause":         ActionPause,
+	"rename":        ActionRename,
+	"resize":        ActionResize,
+	"restart":       ActionRestart,
+	"start":         ActionStart,
+	"stop":          ActionStop,
+	"top":           ActionTop,
+	"unpause":       ActionUnpause,
+	"update":        ActionUpdate,
+}
+
+func (a Action) String() string {
+	for str, act := range stringToAction {
```

Comment:
Why not make it an action to string map, and then do a lookup in to it rather than iterating through the entire map?